MapboxCoreNavigation/RouteController.swift

```diff
@@ -232,7 +232,9 @@ open class RouteController: NSObject {
     }
     
     var userSnapToStepDistanceFromManeuver: CLLocationDistance?
-
+    
+    var simulatedLocationManager: SimulatedLocationManager?
```

Comment:
Can we rename this to tunnelLocationManager or similar to avoid confusion?

Reply:
@frederoni this `simulatedLocationManager` will be used for other context of dead reckoning not just tunnel navigation. I had followed up with @brsbl regarding this simulated navigation effort for dead reckoning.

Reply:
Ok, how about `animatedLocationManager` then? `simulator` should be reserved for the actual simulator in the sdk I think.

Reply:
🤔 maybe we should have a `MapboxLocationManager` to manage all of these real/faux `CLLocationManager` objects? Something to combine the (three now?) native managers (simulated, native, and tunnel) into one concern. There would also be some advantages here in testability, dynamically enabling/disabling route sim, and handoff between one location manager and another, say when entering/exiting a tunnel.

It doesn't have to happen now (or ever), just an idea.
/cc @akitchen

Reply:
@JThramer I concur. We should have a LocationManager to handle the 3 variations of location manager. @frederoni, @1ec5 and I had a chat and we concluded there is definitely candidacy for `ReplayLocationManager`, `SimulatedLocationManager` to be tied to the location manager, `MapboxLocationManager`.  I intend to track this in a new issue ticket for an upcoming PR. Stay tuned 😺

Reply:
I still disagree about this. Names are cheap, and this variable name tips off the developer that this `SimulatedLocationManager` is used for simulation.  While we are simulating, this is particular kind of simulation that we should distinguish from others.

Even if this variable will be used for other types of animation in the future, we can still come up with a better name now that does not lead to confusion.